dvc/command/init.py

```diff
@@ -39,14 +39,14 @@ def run(self):
         from dvc.repo import Repo
 
         try:
-            self.repo = Repo.init(
+            with Repo.init(
```

Comment:
`init` is used in a lot of places in our tests, causing a lot of PermissionErrors  if we are not closing the cache handles.